realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -170,8 +170,10 @@ file(GLOB objectstore_SRC
 # Sync needed Object Store files
 if (build_SYNC)
     file(GLOB objectstore_sync_SRC
-        "object-store/src/sync/*"
-        "object-store/src/sync/impl/*")
+        "object-store/src/results.cpp"
+        "object-store/src/impl/results_notifier.cpp"
```

Comment:
we're using `SyncManager` which pulls the `metadata` dependency from the OS which need the definition of these 2 files to compile.